public/directives/wz-tag-filter/wz-tag-filter.js

```diff
@@ -0,0 +1,238 @@
+/*
+ * Wazuh app - Wazuh search and filter by tags bar
+ * Copyright (C) 2018 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import template from './wz-tag-filter.html';
+import { DataFactory } from '../../services/data-factory';
+import { uiModules } from 'ui/modules';
+
+const app = uiModules.get('app/wazuh', []);
+
+app.directive('wzTagFilter', function () {
+  return {
+    restrict: 'E',
+    scope: {
+      path: '=path',
+      keys: '=keys',
+      queryFn: '&'
+    },
+    controller(
+      $scope,
+      $timeout,
+      apiReq
+    ) {
+      const instance = new DataFactory(
+        apiReq,
+        $scope.path,
+        {}
+      );
+
+      $scope.tagList = [];
+      $scope.groupedTagList = [];
+      $scope.newTag = '';
+      $scope.isAutocomplete = false;
+      $scope.dataModel = [];
+
+      $scope.addTag = (flag = false) => {
+        var input = document.getElementById('wz-search-filter-bar-input');
```

Comment:
- Don't use `var`, use `const` or `let` if that value should be modified.
- Don't use `document` please use `$document` or `$` (jQuery) instead.

Reply:
I think you must use try/catch here